### ISSUE TITLE
A bug when <a> tag is a child of <cite> tag

### ISSUE THREAD

**Issue by ExillustX** [validator/validator#569]
When `<h5>` tag is a child of a `<cite>`:

### Example:
```
<cite>      
    <h5>Julia Sakura</h5>
</cite>
```

### The result shows:

> Error: Element **h5** not allowed as child of element **`<cite>`** in this context. (Suppressing further errors from this subtree.)

which i think the error's result is correct.

But when `<a>` tag is a parent of `<h5>` tag

### Example:

```
<cite>             
 <a>
    <h5>Julia Sakura</h5>
 </a>
</cite>
```
### The result shows:

> Error: Element **h5** not allowed as child of element **`<a>`** in this context. (Suppressing further errors from this subtree.)

Which i think its not correct, heading tags are allowed as a child of `<a>`.

**Comment by cvrebert** [comment]
It's still invalid, but the error message here isn't very clear.

> [4.5.1 The `<a>` element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element)
> Content model:
> **Transparent**, but there must be no interactive content or `<a>` element descendants.

And a [**transparent**](https://html.spec.whatwg.org/multipage/dom.html#transparent) content model means:

> The content model of a transparent element is derived from the content model of its parent element:

So basically, for content model purposes, act like the `<a>` doesn't exist. Thus, the `<h5>` is disallowed because it's not allowed by the content model of `<cite>` (the `<a>`'s parent).

**Comment by sideshowbarker** [comment]
Yeah unfortunately the error message isn’t as helpful as it could be here, but in the current design of the backend code for the validator, it’s very difficult to make it better.

But that said, this kind of case is the reason that error message includes the phrase *“in this context”*; that is, the error message isn’t asserting the `h5` is *never* allowed as a child of an `a` element — instead it’s just saying the specific context in which that particular `h5` occurs isn’t valid.

And the reason it’s not valid is exactly because of what @cvrebert explains in https://github.com/validator/validator/issues/569#issuecomment-341279698.

So, ideally the message would say something like, _“The `h5` element is not allowed in an `a` element when the `a` element is in a `cite` element.”_. But as I said above, in the current design of the backend code for the validator, it’s very difficult to have it emit an error message like that for this case.

It’s a general deficiency in the validator design (not specific to this particular `h5`+`a`+`cite` case), and having it emit the kind of error message I’ve described for these kinds of cases would basically require a fundamental rewrite/replacement of the core validation code in the backend.

**Comment by sideshowbarker** [comment]
I’ve implemented a message clarification for this case, and deployed it.

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%3Bbase64%2CPCFkb2N0eXBlIGh0bWw%2BPGh0bWwgbGFuZz48bWV0YSBjaGFyc2V0PXV0Zi04Pjx0aXRsZT50ZXN0PC90aXRsZT4NCjxjaXRlPiAgICAgICAgICAgICANCiA8YT4NCiAgICA8aDU%2BSnVsaWEgU2FrdXJhPC9oNT4NCiA8L2E%2BDQo8L2NpdGU%2B

Given this document:

```html
    <!doctype html><html lang><meta charset=utf-8><title>test</title>
    <cite>
     <a>
        <h5>Julia Sakura</h5>
     </a>
    </cite>
```
…the checker now emits this message:

> _Error: Element [h5](https://html.spec.whatwg.org/#the-h1,-h2,-h3,-h4,-h5,-and-h6-elements) not allowed as child of element [a](https://html.spec.whatwg.org/#the-a-element) in this context. (Suppressing further errors from this subtree.) Note: The “a” element has a transparent content model; its allowed content is inherited from its parent element._
